tests/cypress/integration/publisher/004-endpoints/08-usage-warning-certificates.spec.js

```diff
@@ -0,0 +1,161 @@
+            cy.get('[data-testid="itest-api-config"]', {timeout: Cypress.config().largeTimeout}).click();
+            cy.get('#left-menu-itemendpoints').click();
+
+            // Expand the general config section to upload certificate
+            cy.get('#http-panel1bh-header').trigger('click');
+            cy.get('#certs-add-btn').click();
+
+            // Select endpoint
+            cy.get('#endpoint-certificate').click();
+            cy.get(`[data-value="${selectedEndpoint}"]`).click({multiple: true});
+
+            // Set alias
+            cy.get('#certificateAlias').click();
+            cy.get('#certificateAlias').type(alias);
+
+            // Select the cert
+            cy.get('[data-testid="cert-upload-btn"]').click();
+            cy.get('input[type="file"]').attachFile(certPath);
+
+            // Click away
+            cy.get('#certificateAlias').click();
+
+            // Save the cert
+            cy.get('#upload-cert-save-btn').click();
+
+            // Logout
+            cy.visit(`/publisher/services/logout`);
+        });
+    })
+
+    beforeEach(function (){
+        cy.loginToPublisher(publisher, password);
+
+        // Visiting endpoints tab of selected API
+        cy.visit(`/publisher/apis/${apiId}/overview`);
+        cy.get('[data-testid="itest-api-config"]', {timeout: Cypress.config().largeTimeout}).click();
+        cy.get('#left-menu-itemendpoints').click();
+
+        // Expand the general config section to upload certificate
+        cy.get('#http-panel1bh-header').trigger('click');
+    })
+
+    it.only("Has correctly working pagination", () => {
+        cy.get('#certificate-usage-btn').click();
+
+        // Test pagination - num of rows 5
+        testPagination(5, random_number);
+
+        // Test pagination - num of rows 10
+        cy.get("#pagination-rows").click();
+        cy.get("#pagination-menu-list>li").eq(1).click();
+        testPagination(10, random_number);
+
+        cy.get('#certificate-usage-cancel-btn').click();
+    });
+
+    it.only("Has correct warning message", () => {
+        // Expected warning message
+        const expectedMessage = `${alias} is used by ${random_number} other APIs`;
+
+        // Test for correct warning message
+        cy.get('#delete-cert-btn').click({force: true});
+        cy.get('#warning-message>p').should('include.text', expectedMessage);
+
+        // Delete certificate
+        cy.get('#delete-cert-confirm-btn').click({force: true});
+    });
+
+    after(function () {
+        apiIds.forEach(id => {
+            Utils.deleteAPI(id);
+        })
+    });
+});
+
+function testPagination(numOfRows, random_number) {
+    cy.get("#certificate-usage-table")
+        .find("tr")
+        .should('have.length', numOfRows + 2); // Plus two is for table header and footer
+
+    let pages = Math.floor(random_number / numOfRows);
+    let numOfRowsInFinalPage = random_number % numOfRows;
+
+    if(pages === 0) {
```

Comment:
Let's fix the formatting in this method

Reply:
Fixed https://github.com/wso2/apim-apps/pull/401